src/velo-fed/BaseFedCCTP.sol

```diff
@@ -31,129 +30,186 @@ interface ICCTP {
     ) external returns (uint64 _nonce);
 }
 
-
 contract BaseFedCCTP {
-    address public chair;
-    address public gov;
-    address public pendingGov;
-    uint public dolaSupply;
-    uint public maxSlippageBpsDolaToUsdc;
-    uint public maxSlippageBpsUsdcToDola;
-
-    uint constant PRECISION = 10_000;
-    uint public constant DOLA_USDC_CONVERSION_MULTI= 1e12;
-
-    IDola public constant DOLA = IDola(0x865377367054516e17014CcdED1e7d814EDC9ce4);
-    IERC20 public constant USDC = IERC20(0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48);
-    IL1ERC20Bridge public constant baseBridge = IL1ERC20Bridge(0x3154Cf16ccdb4C6d922629664174b904d80F2C35);
-    address public constant DOLA_BASE = 0x4621b7A9c75199271F773Ebd9A499dbd165c3191;
-    address public constant USDC_BASE = 0xd9aAEc86B65D86f6A7B5B1b0c42FFA531710b6CA;
-    ICurvePool public curvePool = ICurvePool(0xE57180685E3348589E9521aa53Af0BCD497E884d);
-    ICCTP public constant CCTP = ICCTP(0xBd3fa81B58Ba92a82136038B25aDec7066af3155);
-    uint32 public constant BASE_CCTP_DOMAIN = 6;
-    int128 public dolaIndex = 0;
-    int128 public usdcIndex = 2;
-    address public aeroFarmer;
-
-    event Expansion(uint amount);
-    event Contraction(uint amount);
-
     error OnlyGov();
     error OnlyPendingGov();
     error OnlyChair();
     error CantBurnZeroDOLA();
     error MaxSlippageTooHigh();
-    error DeltaAboveMax();
+    error SlippageTooHigh();
     error SwapMoreDolaThanMinted();
+    error InvalidProxyAddress();
+    error SwapFailed();
+
+    address public chair;
+    address public gov;
+    address public pendingGov;
+    uint256 public dolaSupply;
+    uint256 public maxSlippageBpsDolaToUsdc;
+    uint256 public maxSlippageBpsUsdcToDola;
+    address public exchangeProxy;
+    address public aeroFarmer;
+
+    uint256 public constant PRECISION = 10_000;
+    uint256 public constant DOLA_USDC_CONVERSION_MULTI = 1e12;
+
+    IDola public constant DOLA =
+        IDola(0x865377367054516e17014CcdED1e7d814EDC9ce4);
+    IERC20 public constant USDC =
+        IERC20(0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48);
+    IL1ERC20Bridge public constant baseBridge =
+        IL1ERC20Bridge(0x3154Cf16ccdb4C6d922629664174b904d80F2C35);
+    address public constant DOLA_BASE =
+        0x4621b7A9c75199271F773Ebd9A499dbd165c3191;
+    address public constant USDC_BASE =
+        0xd9aAEc86B65D86f6A7B5B1b0c42FFA531710b6CA;
+    ICCTP public constant CCTP =
+        ICCTP(0xBd3fa81B58Ba92a82136038B25aDec7066af3155);
+    uint32 public constant BASE_CCTP_DOMAIN = 6;
```

Comment:
Should probably make these params immutable constructor params so we can reuse for different superchains.